pkg/controller/backup_deletion_controller_test.go

```diff
@@ -1022,39 +1098,113 @@ func TestDeleteCSIVolumeSnapshots(t *testing.T) {
 	}
 
 	testCases := []struct {
-		name   string
-		backup *velerov1.Backup
+		name       string
+		backupName string
 	}{
 		{
-			name:   "should delete volumesnapshots bound to existing volumesnapshotcontent",
-			backup: builder.ForBackup("velero", "backup1").Result(),
+			name:       "should delete volumesnapshots bound to existing volumesnapshotcontent",
+			backupName: "backup1",
 		},
 		{
-			name:   "should delete volumesnapshots with nil status",
-			backup: builder.ForBackup("velero", "backup2").Result(),
+			name:       "should delete volumesnapshots with nil status",
+			backupName: "backup2",
 		},
 		{
-			name:   "should delete volumesnapshots with nil BoundVolumeSnapshotContentName",
-			backup: builder.ForBackup("velero", "backup3").Result(),
+			name:       "should delete volumesnapshots with nil BoundVolumeSnapshotContentName",
+			backupName: "backup3",
 		},
 		{
-			name:   "should delete volumesnapshots bound to non-existent volumesnapshotcontents",
-			backup: builder.ForBackup("velero", "backup4").Result(),
+			name:       "should delete volumesnapshots bound to non-existent volumesnapshotcontents",
+			backupName: "backup4",
 		},
 		{
-			name:   "should be a no-op when there are no volumesnapshots to delete",
-			backup: builder.ForBackup("velero", "backup-no-vs").Result(),
+			name:       "should be a no-op when there are no volumesnapshots to delete",
+			backupName: "backup-no-vs",
 		},
 	}
 
+	log := velerotest.NewLogger().WithFields(
+		logrus.Fields{
+			"unit-test": "unit-test",
+		},
+	)
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			log := velerotest.NewLogger().WithFields(
-				logrus.Fields{
-					"unit-test": "unit-test",
-				},
-			)
-			errs := deleteCSIVolumeSnapshots(tc.backup, fakeSharedInformer.Snapshot().V1beta1().VolumeSnapshots().Lister(), fakeClient.SnapshotV1beta1(), log)
+			errs := deleteCSIVolumeSnapshots(tc.backupName, fakeSharedInformer.Snapshot().V1beta1().VolumeSnapshots().Lister(), fakeClient.SnapshotV1beta1(), log)
+			assert.Empty(t, errs)
+		})
+	}
+}
+
+func TestDeleteCSIVolumeSnapshotContents(t *testing.T) {
+	retainVSC := snapshotv1beta1api.VolumeSnapshotContent{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "retainVSC",
+			Labels: map[string]string{
+				velerov1.BackupNameLabel: "backup1",
+			},
+		},
+		Spec: snapshotv1beta1api.VolumeSnapshotContentSpec{
+			DeletionPolicy: snapshotv1beta1api.VolumeSnapshotContentRetain,
+		},
+	}
+	deleteVSC := snapshotv1beta1api.VolumeSnapshotContent{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "deleteVSC",
+			Labels: map[string]string{
+				velerov1.BackupNameLabel: "backup2",
+			},
+		},
+		Spec: snapshotv1beta1api.VolumeSnapshotContentSpec{
+			DeletionPolicy: snapshotv1beta1api.VolumeSnapshotContentDelete,
+		},
+	}
+
+	nothingVSC := snapshotv1beta1api.VolumeSnapshotContent{
```

Comment:
Would it work define the VSCs in the test cases, then add them to the client and informer in the test loop?

That way, all the data for the test case is collected in the `testCases` slice, at the expense of making the contents of `t.Run` longer. It would be easier to edit the case data in case more get added or changes are needed, though.

I have a similar comment for the `TestSetVolumeSnapshotContentDeletionPolicy` tests above

Reply:
done!